### DISCUSSION TITLE
Add missing import from the editor

### DISCUSSION THREAD

**Discussion by huypham50** [zed-industries/zed#13115] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

Sorry if I missed this somewhere but I don't see a way to import missing variable in Zed.

<img width="639" alt="Screenshot 2024-06-15 at 3 05 28 PM" src="https://github.com/zed-industries/zed/assets/29948820/73f27e79-8d07-4ef6-8644-c7c8948f2c54">

Workaround right now is to import while typing out like this:

<img width="700" alt="Screenshot 2024-06-15 at 3 06 55 PM" src="https://github.com/zed-industries/zed/assets/29948820/22c3faba-c00c-469d-8c4b-4368f0d03cd8">

**Comment by Moshyfawn** [comment]
Try hitting that ⚡️ icon in the editor's gutter on the left: 

<img width="1246" alt="image" src="https://github.com/zed-industries/zed/assets/16290753/eeeb0a9d-7de1-42dc-8b4a-57dae365c396">

**Reply by iatomic1** [reply]
is there a way to add keybind for it? smthn like `space c u` to add all missing imports

**Comment by aqc-carlodri** [comment]
Looking for this as well for Python. Lack of this functionality forces me to type the imports manually....which drives me back to VSCode.

**Reply by aqc-carlodri** [reply]
turns ot that this was due to the fact that I was trying out ty as a language server.... basedpyright offers auto-import

**Reply by aqc-carlodri** [reply]
....and because I was using pixi to manage my python env. If you specify the python interpreter path like this: https://docs.astral.sh/ty/reference/configuration/#python, then ty import actions magically work!

**Reply by aqc-carlodri** [reply]
hmm, not completely true, after some testing ty import actions appear only in some cases, not sure why.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
